src/screens/game_map.py

```diff
@@ -0,0 +1,646 @@
+        image = pygame.Surface(size)
+        Sprite(pos, image, z=layer, name=name).add(groups)
+
+        if SETUP_PATHFINDING:
+            self._add_pf_matrix_collision(
+                (obj.x, obj.y), (obj.width, obj.height)
+            )
+
+    def _setup_collideable_object(
+            self,
+            pos: tuple[int, int],
+            obj: TiledObject,
+            layer: Layer,
+            groups: tuple[pygame.sprite.Group, ...] | pygame.sprite.Group
+    ):
+        """
+        Create a new collideable Sprite from the given TiledObject.
+        If this map's MapObjects instance doesn't contain this object's GID,
+        this method will raise an exception
+        """
+        CollideableMapObject(
+            pos, self._map_objects[obj.gid], z=layer
+        ).add(groups)
+
+        if SETUP_PATHFINDING:
+            self._add_pf_matrix_collision(
+                (obj.x, obj.y), (obj.width, obj.height)
+            )
+
+    def _setup_tree_object(
+            self,
+            pos: tuple[int, int],
+            obj: TiledObject,
+            groups: tuple[pygame.sprite.Group, ...] | pygame.sprite.Group,
+    ):
+        """
+        Create a new collideable Sprite from the given TiledObject.
+        This Sprite will be created from the Tree class and will take the apple
+        and stump assets from self.frames
+        :param pos: Position of Sprite (x, y)
+        :param obj: TiledObject to create the Tree from
+                    (the obj.image will be used as Tree image)
+        :param groups: Groups the Sprite should be added to
+        """
+        props = obj.properties
+        if props.get("type") == "tree":
+            if props.get("size") == "medium" and props.get("breakable"):
+                fruit = props.get("fruit_type")
+                if fruit == "no_fruit":
+                    fruit_type, fruit_frames = None, None
+                else:
+                    fruit_type = InventoryResource.from_serialised_string(fruit)
+                    fruit_frames = self.frames['level']['objects'][fruit]
+                stump_frames = self.frames['level']['objects']['stump']
+
+                tree = Tree(
+                    pos, self._map_objects[obj.gid],
+                    (
+                        self.all_sprites,
+                        self.collision_sprites,
+                        self.tree_sprites
+                    ),
+                    obj.name,
+                    fruit_frames,
+                    fruit_type,
+                    stump_frames,
+                    self.drops_manager
+                )
+                # we need a tree surf without fruits
+                tree.image = self.frames['level']['objects']['tree']
+                tree.surf = tree.image
+
+        if SETUP_PATHFINDING:
+            self._add_pf_matrix_collision(
+                (obj.x, obj.y), (obj.width, obj.height)
+            )
+
+    def _setup_player_point(
```

Comment:
This is a nitpick, but "point" seems a bit too vague, "warp" would work better

Reply:
Hmm I don't know, somehow "warp" sounds a bit off when walking from one map to another. How about "travel", "transit" or "traverse point"? The name should also somehow imply that it means spawnpoints as well.

Reply:
Warp is game lingo, actually, so even if the maps are meant to be linked to each other, that shouldn't be too weird

Reply:
Good to know, then you're probably right 😅